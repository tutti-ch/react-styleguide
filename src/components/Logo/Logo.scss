@import "../../styles/main";

.logo {
  display: flex;
  align-items: center;
  height: 2.2rem;
  width: 10rem;
  @include z-index("overlay", "ground");

  .symbolBox {
    position: relative;
    margin-right: $size-xxs;
    display: flex;
    align-items: center;

    .symbolSquare {
      height: 2.6rem;
      transform-origin: 1.8rem 1.8rem;
      transition: $transition-m;
    }

    .symbolT {
      position: absolute;
      bottom: 0.1rem;
      height: 1.8rem;
      left: 0.7rem;
    }
  }

  .font {
    height: 1.6rem;
    margin-top: 0.1rem;
  }

  &:hover {
    .symbolSquare {
      cursor: pointer;
      transform: rotate(-15deg);
      opacity: 0.8;
    }

    .font {
      opacity: 0.9;
    }
  }
}

.fallback {
  display: none;
}

@include ifInternetExplorer {
  .logo {
    width: auto;
  }

  .symbolBox,
  .font {
    display: none !important;
  }

  .fallback {
    display: flex;
  }
}
