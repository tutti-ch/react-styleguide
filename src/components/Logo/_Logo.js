import React from "react";
import PropTypes from "prop-types";

import classes from "./Logo.scss";
import Image from "../Image/Image";
import TuttiSymbolSquare from "./assets/logo-square.svg";
import TuttiSymbolT from "./assets/logo-t-white.svg";
import TuttiFont from "./assets/logo-font.svg";

const Logo = ({ className }) => (
  <div className={classes.logo}>
    <div className={classes.symbolBox}>
      <Image
        className={classes.symbolSquare}
        src={TuttiSymbolSquare}
        alt="tutti.ch"
      />
      <Image className={classes.symbolT} src={TuttiSymbolT} alt="tutti.ch" />
    </div>
    <Image className={classes.font} src={TuttiFont} alt="tutti.ch" />
  </div>
);

Logo.propTypes = {
  className: PropTypes.string
};

export default Logo;
