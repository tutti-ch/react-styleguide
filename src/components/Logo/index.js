export { default } from "./_Logo.js";
