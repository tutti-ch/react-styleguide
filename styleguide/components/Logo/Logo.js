import React from 'react'
import TuttiLogo from './assets/logo.svg'
import classes from './Logo.scss'

const Logo = () => {
  return (
      <img className={classes.logoBox} src={TuttiLogo} alt='tutti.ch logo' />
  )
}

export default Logo
