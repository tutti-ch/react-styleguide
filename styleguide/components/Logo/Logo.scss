.logoBox {
  height: 4rem;
}
