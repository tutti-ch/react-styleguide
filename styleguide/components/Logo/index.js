export { default } from './Logo.js'
