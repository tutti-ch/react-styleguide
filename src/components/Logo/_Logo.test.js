import React from "react";
import Logo from "./_Logo";
import { mount } from "enzyme";

describe("(Components) Logo", () => {
    test("should match the snapshot", () => {
        const comp = mount(<Logo />);
        expect(comp).toMatchSnapshot();
    })
})