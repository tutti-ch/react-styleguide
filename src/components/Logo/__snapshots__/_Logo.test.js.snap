// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) Logo should match the snapshot 1`] = `
<Logo>
  <div>
    <div>
      <div>
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
    <div>
      <svg
        height="34px"
        id="Layer_1"
        viewBox="0 0 158 34.1"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M116.9 26.5c-2.1 0-3.8 1.7-3.8 3.8s1.7 3.8 3.8 3.8 3.8-1.7 3.8-3.8-1.7-3.8-3.8-3.8zM12.6 3h-1c-.2 0-.4.1-.6.2L.2 12.9c-.1.1-.2.3-.2.5v1.7c0 .4.3.8.7.8h3.4v9.3c0 3.2.5 4.9 1.8 6.2 1.5 1.5 4.3 2.5 7.4 2.5 3 0 5.4-.7 7.8-2.4.2-.1.3-.4.3-.6v-5.3c0-.3-.1-.6-.4-.7-.3-.1-.6-.1-.8 0-1.4.7-2.7 1.1-4 1.1-1.9 0-2.7-1-2.7-3.1v-7.1H20c.4 0 .8-.3.8-.7v-4.4c0-.4-.3-.8-.7-.8h-6.7V3.7c0-.3-.4-.7-.8-.7zm35.7 7h-7.7c-.4 0-.8.3-.8.7v13.2c-1.2 1.7-2.2 2.4-3.6 2.4-2.4 0-2.7-1.8-2.7-3.6v-12c0-.4-.3-.8-.7-.8H25c-.4 0-.8.3-.8.7V25c0 2.7.8 8.9 8 8.9 3.2 0 5.8-1.3 8-4 .7 3 2.3 4 6 4 1.2 0 2.4-.1 3.5-.3.3-.1.5-.3.6-.5l1.7-4.6c.1-.2.1-.6-.1-.7-.1-.2-.4-.3-.6-.3h-.1c-1.8 0-2.1-1.6-2.1-2.6v-14c0-.5-.4-.9-.8-.9zm39.5-7h-1c-.2 0-.4.1-.6.2l-10.8 9.5c-.1.1-.2.4-.2.6V15c0 .4.3.8.7.8h3.4v9.3c0 3.2.5 4.9 1.8 6.2 1.5 1.5 4.3 2.5 7.4 2.5 3 0 5.4-.7 7.8-2.4.2-.1.3-.4.3-.6v-5.3c0-.3-.1-.6-.4-.7-.3-.1-.6-.1-.8 0-1.4.7-2.7 1.1-4 1.1-1.9 0-2.7-1-2.7-3.1v-7.1h6.7c.4 0 .8-.3.8-.7v-4.4c0-.4-.3-.8-.7-.8h-6.7V3.6c-.3-.3-.6-.6-1-.6zm44.9 13.9c-6.3 0-9.1 4.3-9.1 8.6 0 4.9 3.8 8.4 9.3 8.4 1.7 0 2.9-.2 4.7-.8.3-.1.6-.4.6-.7v-3.5c0-.2-.1-.5-.3-.6-.1-.1-.3-.1-.5-.2-.1 0-.2 0-.2.1-1.2.4-2 .6-3 .6-2.4 0-4-1.3-4-3.1 0-1.9 1.5-3.2 3.9-3.2 1.1 0 2.1.2 3.1.6.1 0 .2.1.2.1.2 0 .3-.1.5-.2s.3-.4.3-.6v-3.7c0-.4-.3-.7-.6-.8-1.6-.8-3.2-1-4.9-1zm13.7-4.4h-4.9c-.4 0-.8.3-.8.7v19.7c0 .4.3.8.7.8h5c.4 0 .8-.3.8-.7v-9.2c.7-1 1.6-1.6 2.5-1.6 1.7 0 1.8 1.7 1.8 2.4v8.2c0 .4.3.8.7.8h5c.4 0 .8-.3.8-.7V23c0-5.3-3.6-6.1-5.7-6.1-2.2 0-3.7.7-5.3 2.3v-6.1c.2-.3-.1-.6-.6-.6zM108.8 10h-7.7c-.4 0-.8.3-.8.7v22.1c0 .4.3.8.7.8h7.8c.4 0 .8-.3.8-.7V10.8c0-.4-.4-.7-.8-.8zm-3.9-10c-2.3 0-4.2 1.8-4.2 4.2v.1c0 2.3 1.8 4.2 4.2 4.2h.1c2.3 0 4.2-1.9 4.2-4.2S107.3 0 104.9 0zM64.5 3h-1c-.2 0-.4.1-.6.2l-10.8 9.6c-.1.1-.2.4-.2.6v1.7c0 .4.3.8.7.8H56v9.3c0 3.2.5 4.9 1.8 6.2 1.5 1.5 4.3 2.5 7.4 2.5 3 0 5.4-.7 7.8-2.4.2-.1.3-.4.3-.6v-5.3c0-.3-.1-.6-.4-.7-.3-.1-.6-.1-.8 0-1.4.7-2.7 1.1-4 1.1-1.9 0-2.7-1-2.7-3.1v-7.1H72c.4 0 .8-.3.8-.7v-4.4c0-.4-.3-.8-.7-.8h-6.7V3.7c-.1-.4-.5-.7-.9-.7z"
        />
      </svg>
    </div>
  </div>
</Logo>
`;
